### PR TITLE
TEST UNIT

### PR DESCRIPTION
**Pull request by 3dawy**
add uni test folder to test motor module

### CODE REVIEW

File: Static_Design/UNI TEST/Motor_Test/Motor_Test.c [hunk 1]
```diff
@@ -0,0 +1,608 @@
...
+		printf("Please enter test number: ");
+
+		scanf("%d",&au8_TestNumber);
+
+		switch (au8_TestNumber)
+		{
+		case 1:
+
+			gu8_PWM_Init_Return = E_OK;
+			gu8_DIO_Init_Return = E_OK;
```
Comment (YehiaSprints) [inline Static_Design/UNI TEST/Motor_Test/Motor_Test.c]: Does Motor_Init function check if the PWM_init and DIO_init are Ok or not?

File: Static_Design/UNI TEST/Motor_Test/Motor_Test.c [hunk 2]
```diff
@@ -0,0 +1,608 @@
...
+	{
+
+		printf("Please enter test number: ");
+
+		scanf("%d",&au8_TestNumber);
+
+		switch (au8_TestNumber)
+		{
+		case 1:
+			gu8_PWM_Start_Return = E_OK;
```
Comment (YehiaSprints) [inline Static_Design/UNI TEST/Motor_Test/Motor_Test.c]: Where gu8_PWM_Start_Return variable is used?

File: Static_Design/UNI TEST/Motor_Test/PWM_Mock.c [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+#include "PWM.h"
+
+
+extern uint8_t gu8_PWM_Init_Return;
+extern uint8_t gu8_PWM_Start_Return;
+
+
+ERROR_STATUS Pwm_Init(Pwm_Cfg_s *Pwm_Cfg)
+{
+	return gu8_PWM_Init_Return;
```
Comment (YehiaSprints) [inline Static_Design/UNI TEST/Motor_Test/PWM_Mock.c]: The MOCK should return OK all the time.

File: Static_Design/UNI TEST/Motor_Test/Motor_Test.c (hunk 2 above)
Comment (3dawy) [reply Static_Design/UNI TEST/Motor_Test/Motor_Test.c]: PWM start mock function

File: Static_Design/UNI TEST/Motor_Test/PWM_Mock.c (hunk 3 above)
Comment (3dawy) [reply Static_Design/UNI TEST/Motor_Test/PWM_Mock.c]: I want to know how the function will behave against error in the called function

**Comment by YehiaSprints** [comment]
Please merge and close.

**Pull request merged**